render-wasm/src/main.rs

```diff
@@ -96,10 +97,17 @@ pub extern "C" fn set_view_xy(x: f32, y: f32) {
 }
 
 #[no_mangle]
-pub extern "C" fn use_shape(a: u32, b: u32, c: u32, d: u32) {
+pub extern "C" fn use_shape_rect(a: u32, b: u32, c: u32, d: u32) {
     let state = unsafe { STATE.as_mut() }.expect("got an invalid state pointer");
     let id = uuid_from_u32_quartet(a, b, c, d);
-    state.use_shape(id);
+    state.use_shape(id, Kind::Rect);
+}
+
+#[no_mangle]
+pub extern "C" fn use_shape_circle(a: u32, b: u32, c: u32, d: u32) {
+    let state = unsafe { STATE.as_mut() }.expect("got an invalid state pointer");
+    let id = uuid_from_u32_quartet(a, b, c, d);
+    state.use_shape(id, Kind::Circle);
```

Comment:
Question: why do we need to do this (and then pass the kind to every shape-related method)? We're already storing the right kind when creating the shape, right?